src/hamiltonian.jl

```diff
@@ -23,7 +23,8 @@ struct DualValue{V<:AbstractScalarOrVec{<:AbstractFloat}, G<:AbstractVecOrMat{<:
     end
 end
 
-Base.similar(dv::DualValue{<:AbstractVector}) = DualValue(similar(dv.value), similar(dv.gradient))
+Base.similar(dv::DualValue{<:AbstractVecOrMat{T}}, args...) where {T<:AbstractFloat} = 
```

Comment:
Do we really need the `argus..` argument? It does not to be used anywhere.

Reply:
it's a prepare for supporting `similar(z, 10)`, which will initialise a phase point with an extra dimension of length 10. But I guess I should introduce it later. Let me remove it for now.